metricbeat-tests/runner_test.go

```diff
@@ -54,78 +58,85 @@ func TestMain(m *testing.M) {
 	os.Exit(status)
 }
 
-func cleanUpOutputs() {
-	dir, _ := os.Getwd()
-
-	pattern := dir + "/outputs/*.metrics*"
-
-	files, err := filepath.Glob(pattern)
-	if err != nil {
-		log.WithFields(log.Fields{
-			"error": err,
-			"path":  pattern,
-		}).Fatal("Cannot retrieve outputs :(")
+// assertHitsArePresent returns an error if no hits are present
+func assertHitsArePresent(hits map[string]interface{}, q ElasticsearchQuery) error {
+	hitsCount := len(hits["hits"].(map[string]interface{})["hits"].([]interface{}))
+	if hitsCount == 0 {
+		return fmt.Errorf(
+			"There aren't documents for %s-%s on Metricbeat index",
+			q.EventModule, q.ServiceVersion)
 	}
 
-	for _, f := range files {
-		err := os.Remove(f)
-		if err != nil {
-			log.WithFields(log.Fields{
-				"error": err,
-				"path":  f,
-			}).Fatal("Cannot remove output file :(")
-		}
-	}
+	return nil
 }
 
-func metricbeatOutputsMetricsToTheFile(fileName string) error {
-	dir, _ := os.Getwd()
-
-	filePath := dir + "/outputs/" + fileName
-
-	foundChannel := make(chan bool, 1)
-
-	go fileChecker(foundChannel)
-	initialLoops := 15
-	loops := initialLoops
-	seconds := 2
-
-	for {
-		exists := fileExists(filePath)
-		if exists {
-			foundChannel <- true
-			close(foundChannel)
-			break
+// assertHitsDoNotContainErrors returns an error if any of the returned entries contains
+// an "error.message" field in the "_source" document
+func assertHitsDoNotContainErrors(hits map[string]interface{}, q ElasticsearchQuery) error {
+	for _, hit := range hits["hits"].(map[string]interface{})["hits"].([]interface{}) {
```

Comment:
Same comment here about unchecked type conversions.

Reply:
Will work on a DSL to simplify its usage